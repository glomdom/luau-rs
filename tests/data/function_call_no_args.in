fn main() -> i32 {
    test_fn()
}

fn test_fn() -> i32 {
    2
}