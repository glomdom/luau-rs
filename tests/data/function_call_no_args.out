Block(
    Block {
        statements: [
            Function(
                Function {
                    name: "main",
                    params: [],
                    ret_type: Some(
                        LuauType {
                            type_name: "number",
                        },
                    ),
                    body: Block(
                        Block {
                            statements: [
                                Return(
                                    Return {
                                        value: Some(
                                            Call(
                                                Call {
                                                    func: "test_fn",
                                                    args: [],
                                                },
                                            ),
                                        ),
                                    },
                                ),
                            ],
                        },
                    ),
                },
            ),
            Function(
                Function {
                    name: "test_fn",
                    params: [],
                    ret_type: Some(
                        LuauType {
                            type_name: "number",
                        },
                    ),
                    body: Block(
                        Block {
                            statements: [
                                Return(
                                    Return {
                                        value: Some(
                                            Value(
                                                Value {
                                                    value: "2",
                                                },
                                            ),
                                        ),
                                    },
                                ),
                            ],
                        },
                    ),
                },
            ),
        ],
    },
)